opencog/ure/forwardchainer/ForwardChainer.cc

```diff
@@ -411,6 +412,28 @@ HandleSet ForwardChainer::apply_rule(const Rule& rule)
 		AtomSpace derived_rule_as(&ref_as);
 		Handle rhcpy = derived_rule_as.add_atom(rule.get_rule());
 
+		HandleSeq clauses = rule.get_clauses();
+		Handle vdecl = rule.get_vardecl();
+		HandleSet varset1;
+		HandleSet varset2;
+
+		if (vdecl != Handle::UNDEFINED)
+		{
+			if (VariableListCast(vdecl))
+				varset1 = VariableListCast(vdecl)->get_variables().varset;
+			else
+				varset1.insert(vdecl);
+
+			for (auto var : varset1)
+				if (var->get_type() == TYPED_VARIABLE_LINK)
+					varset2.insert(var->getOutgoingAtom(0));
```

Comment:
I don't think you need that, `varset` elements are always either `VariableNode` or `GlobNode` nothing else.

Reply:
Maybe if i get the variables in a another way this won't be an issue but if i get them through the vdecl i have to remove TypedVariableLinks as this was causing some Unit Tests to fail.

Reply:
Oh, indeed, you need that in case you have only one typed variable. Though if you directly get  `Variables` from `Rule::_rule` then it shouldn't been needed.